library/rendering/entity_rendering/src/main/java/org/quiltmc/qsl/rendering/entity/api/client/ArmorRenderingRegistry.java

```diff
@@ -0,0 +1,84 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.rendering.entity.api.client;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.client.render.entity.model.BipedEntityModel;
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.ItemConvertible;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.Identifier;
+
+import org.quiltmc.qsl.base.api.event.Event;
+import org.quiltmc.qsl.rendering.entity.impl.client.ArmorRenderingRegistryImpl;
+
+@Environment(EnvType.CLIENT)
+public final class ArmorRenderingRegistry {
+	private ArmorRenderingRegistry() {
+		throw new RuntimeException("ArmorRenderingRegistry only contains static declarations.");
+	}
+
+	public static final Identifier DEFAULT_PHASE = Event.DEFAULT_PHASE;
+
+	public static void registerTextureProvider(@NotNull Identifier phaseIdentifier,
+			@NotNull TextureProvider provider, @NotNull ItemConvertible... items) {
+		ArmorRenderingRegistryImpl.TEXTURE_PROVIDERS.addProvider(phaseIdentifier, provider, items);
+	}
+
+	public static void addTextureProviderPhaseOrdering(@NotNull Identifier firstPhase, @NotNull Identifier secondPhase) {
+		ArmorRenderingRegistryImpl.TEXTURE_PROVIDERS.addPhaseOrdering(firstPhase, secondPhase);
+	}
+
+	public static void registerTextureProvider(@NotNull TextureProvider provider, @NotNull ItemConvertible... items) {
+		registerTextureProvider(DEFAULT_PHASE, provider, items);
+	}
+
+	public static void registerModelProvider(@NotNull Identifier phaseIdentifier,
+			@NotNull ModelProvider provider, @NotNull ItemConvertible... items) {
+		ArmorRenderingRegistryImpl.MODEL_PROVIDERS.addProvider(phaseIdentifier, provider, items);
+	}
+
+	public static void addModelProviderPhaseOrdering(@NotNull Identifier firstPhase, @NotNull Identifier secondPhase) {
+		ArmorRenderingRegistryImpl.MODEL_PROVIDERS.addPhaseOrdering(firstPhase, secondPhase);
+	}
+
+	public static void registerModelProvider(@NotNull ModelProvider provider, @NotNull ItemConvertible... items) {
+		registerModelProvider(DEFAULT_PHASE, provider, items);
+	}
+
+	@Environment(EnvType.CLIENT)
+	public interface TextureProvider {
+		/**
+		 * @param useSecondTexture {@code false} to use helmet/chestplate texture, {@code true} to use leggings/boots texture
+		 * @param suffix           suffix to append to the end of the texture name
+		 */
+		@NotNull Identifier getArmorTexture(@NotNull Identifier texture,
+				@NotNull LivingEntity entity, @NotNull ItemStack stack, @NotNull EquipmentSlot slot,
+				boolean useSecondTexture, @Nullable String suffix);
+	}
+
+	@Environment(EnvType.CLIENT)
+	public interface ModelProvider {
+		@NotNull BipedEntityModel<LivingEntity> getArmorModel(@NotNull BipedEntityModel<LivingEntity> model,
```

Comment:
With the way how those are like, wrapping, maybe have

```
method (
  params
);
```

That's extremely nitpicky and subjective so if anyone wanna object feel free.